code/modules/atmospherics/FEA_fire.dm

```diff
@@ -91,196 +90,200 @@
+	var/just_spawned = FALSE
+	var/bypassing = FALSE
 	var/catalyst_active = FALSE
 
-	New()
-		..()
-		START_TRACKING
-		set_dir(pick(cardinal))
+/obj/hotspot/New()
+	..()
+	START_TRACKING
+	set_dir(pick(cardinal))
 #ifndef HOTSPOT_MEDIUM_LIGHTS
-		light = new /datum/light/point
-		light.set_brightness(0.5,queued_run = 1)
-		light.attach(src)
-		// note: light is left disabled until the color is set
+	light = new /datum/light/point
+	light.set_brightness(0.5,queued_run = TRUE)
+	light.attach(src)
+	// note: light is left disabled until the color is set
 #endif
 
-	disposing()
-		STOP_TRACKING
+/obj/hotspot/disposing()
+	STOP_TRACKING
 #ifndef HOTSPOT_MEDIUM_LIGHTS
-		light.disable(queued_run = 1)
+	light.disable(queued_run = TRUE)
 #endif
-		var/turf/simulated/floor/location = loc
-		if (istype(location))
-			location.active_hotspot = null
-		..()
-
-	// now this is ss13 level code
-	proc/set_real_color()
-		var/input = temperature / 100
-
-		var/red
-		if (input <= 66)
-			red = 255
-		else
-			red = input - 60
-			red = 329.698727446 * (red ** -0.1332047592)
-		red = clamp(red, 0, 255)
-
-		var/green
-		if (input <= 66)
-			green = max(0.001, input)
-			green = 99.4708025861 * log(green) - 161.1195681661
-		else
-			green = input - 60
-			green = 288.1221695283 * (green ** -0.0755148492)
-		green = clamp(green, 0, 255)
-
-		var/blue
-		if (input >= 66)
-			blue = 255
+	var/turf/simulated/floor/location = loc
+	if (location.turf_flags & IS_TYPE_SIMULATED)
+		location.active_hotspot = null
+	..()
+
+// now this is ss13 level code
+/// Converts our temperature into an approximate color based on blackbody radiation.
+/obj/hotspot/proc/set_real_color()
+	var/input = temperature / 100
+
+	var/red
+	if (input <= 66)
+		red = 255
+	else
+		red = input - 60
+		red = 329.698727446 * (red ** -0.1332047592)
+	red = clamp(red, 0, 255)
+
+	var/green
+	if (input <= 66)
+		green = max(0.001, input)
+		green = 99.4708025861 * log(green) - 161.1195681661
+	else
+		green = input - 60
+		green = 288.1221695283 * (green ** -0.0755148492)
+	green = clamp(green, 0, 255)
+
+	var/blue
+	if (input >= 66)
+		blue = 255
+	else
+		if (input <= 19)
+			blue = 0
 		else
-			if (input <= 19)
-				blue = 0
-			else
-				blue = input - 10
-				blue = 138.5177312231 * log(blue) - 305.0447927307
-		blue = clamp(blue, 0, 255)
+			blue = input - 10
+			blue = 138.5177312231 * log(blue) - 305.0447927307
+	blue = clamp(blue, 0, 255)
 
-		color = rgb(red, green, blue) //changing obj.color is not expensive, and smooths imperfect light transitions
+	color = rgb(red, green, blue) //changing obj.color is not expensive, and smooths imperfect light transitions
 
-		// dear lord i apologuize for this conditional which i am about to write and commit. please be with the starving pygmies down in new guinea amen
+	// dear lord i apologuize for this conditional which i am about to write and commit. please be with the starving pygmies down in new guinea amen
```

Comment:
apologise*

Reply:
zewaka was feeling really apologuietic you see

Reply:
understandable. yet